app/controllers/project/donate.js

```diff
@@ -25,26 +25,26 @@ export default Controller.extend({
   shouldCreateCustomer: not('stripeCustomerCreated'),
 
   actions: {
-    saveCard(cardParams) {
+    saveAndDonate(amount, cardParams) {
       this._clearErrors();
+      this._updateIsLoading(true);
 
       return this._createCreditCardToken(cardParams)
                  .then((stripeResponse) => this._createCardForPlatformCustomer(stripeResponse))
+                 .then((stripeCard) => this._createSubscription(amount, stripeCard))
+                 .then(() => this._transitionToThankYou())
                  .catch((reason) => this._handleError(reason))
-                 .finally(() => this._updateAddingCardState(false));
+                 .finally(() => this._updateIsLoading(false));
```

Comment:
I could extract the first part into a function to avoid repeating, but really, that would just make it more complicated. This is more explicit, IMO.